include/adc.h

```diff
@@ -27,119 +27,152 @@ extern "C" {
  */
 
 /**
- * @brief ADC driver Sequence entry
- *
- * This structure defines a sequence entry used
- * to define a sample from a specific channel.
+ * @brief ADC sampling definition structure.
  */
-struct adc_seq_entry {
-	/** Clock ticks delay before sampling the ADC. */
-	s32_t sampling_delay;
-
-	/** Buffer pointer where the sample is written.*/
-	u8_t *buffer;
+struct adc_sampling {
+	/** Clock ticks delay before starting the sampling (counted from the
+	 *  start of the previous sampling in the sequence, if any). */
+	u32_t delay;
 
-	/** Length of the sampling buffer.*/
-	u32_t buffer_length;
+	/** Pointer to a buffer where the samples are to be written.
+	 *  Each sample value is sign-extended to 16 bits. */
+	s16_t *buffer;
 
-	/** Channel ID that should be sampled from the ADC */
-	u8_t channel_id;
+	/** Channels to be used for getting consecutive samples. */
+	const u8_t *channel_ids;
 
-	u8_t stride[3];
+	/** Number of samples to read in the sampling. */
+	u8_t sample_count;
 };
 
 /**
- * @brief ADC driver Sequence table
- *
- * This structure defines a list of sequence entries
- * used to execute a sequence of samplings.
+ * @brief ADC sampling definition structure.
  */
-struct adc_seq_table {
-	/* Pointer to a sequence entry array. */
-	struct adc_seq_entry *entries;
+struct adc_channel_cfg {
+	u8_t input_positive;
+	u8_t input_negative;
+	/**
+	 *  A bit field with the following parts:
+	 *  [5 :  0] - acquisition time, in microseconds
```

Comment:
6 bits for the acquisition time looks a bit short, as 63 microseconds may be too small if you sample a high-impedance source or your ADC runs on a very slow clock.
On the other hand, 1us is rather high if you go high speed. For the STM32L4A6, I can setup Tacq as low as 30 nanoseconds.
So maybe we need a more flexible approach to specify the conversion time?